extensions/amp-instagram/0.1/amp-instagram.js

```diff
@@ -76,16 +82,45 @@ class AmpInstagram extends AMP.BaseElement {
   }
 
   /** @override */
-  prerenderAllowed() {
-    return true;
+  prerenderCallback() {
+    const image = new Image();
+    // This will redirect to the image URL. By experimentation this is
+    // always the same URL that is actually used inside of the embed.
+    image.src = 'https://www.instagram.com/p/' +
+        encodeURIComponent(this.shortcode_) + '/media/?size=l';
+    image.width = this.element.getAttribute('width');
+    image.height = this.element.getAttribute('height');
+    setStyles(image, {
+      'object-fit': 'cover',
+    });
+    const wrapper = document.createElement('wrapper');
+    // This makes the non-iframe image appear in the exact same spot
+    // where it will be inside of the iframe.
+    setStyles(wrapper, {
+      'position': 'absolute',
+      'top': '48px',
+      'bottom': '48px',
+      'left': '8px',
+      'right': '8px',
+    });
+    wrapper.appendChild(image);
+    this.placeholderWrapper_ = wrapper;
+    this.applyFillContent(image);
+    this.element.appendChild(wrapper);
```

Comment:
Yep, as mentioned elsewhere, progressive image loading is a good idea to preserve. Let's keep and make a cannon.